### ISSUE TITLE
Support for local satellite reference coordinate system

### ISSUE THREAD

**Issue by matijabrumat** [skyfielders/python-skyfield#492]
In skyfield I want to calculate alt-az distance from one reference satellite to next (for antenna pointing simulation). 
My current workaround is to create the reference observer at sat1 height and calculate the alt-az angles:

```
    sat_observer = sat_ref.subpoint()
    ref_PoV = Topos(sat_observer.latitude, sat_observer.longitude, elevation_m=sat_observer.elevation.m)
    
    satellite = Sat(name)
    orbit = (satellite - ref_PoV).at(time[0])
    el, az, distance = orbit.altaz()
```

Code is able to calculate relative pointing angles at one time instance. This method is not working for absolute values as the alt-az reference always points to Earth north (the local observer coordinate system at satellite panel will rotate around alt axis in the next time instant).

It would be nice to have support for local coordinate system (LVLH frame) at satellite in skyfield to obtain the alt-az angles respective to this LVLH frame. 

Further discussion can be found on stackexchange [here](https://space.stackexchange.com/questions/48796/local-observer-coordinate-system-at-satellite-panel-lvlh-coordinate-system).

**Comment by brandon-rhodes** [comment]
For the moment it might work for you to simply apply a rotation — which axis in particular do you mean by “alt axis”? Do you mean what might also be called the zenith-to-nadir axis? So your satellite is spinning always on an axis pointing down at the Earth? (Though I can't quite imagine how that works because of angular momentum.)

**Comment by matijabrumat** [comment]
Regarding alt-axis: yes it is zenith-to-nadir axis, i kept the nomenclature
from the skyfield function call (.altaz). In mechanisms it is usually named
elevation.

Satellite is not spinning, it is fixed, but the antenna pointing mechanism
has 2 articulated axes, as shown in Fig.1 here:
https://www.researchgate.net/publication/251976475_Satellite_antenna_control_Design_and_performance_validation_under_given_TPF

In order to calculate the alt-az (or el-az) trajectory (antenna direction
alignment) between our reference, where antena sits, and target satellite
we need a local reference coordinate system on the satellite panel. Such
reference allows to calculate absolute alt-az (or el-az) view angles at
arbitrary time.

I hope it is any clearer now.


V V pet., 27. nov. 2020 ob 16:41 je oseba Brandon Rhodes <
notifications@github.com> napisala:

**Comment by brandon-rhodes** [comment]
Yes, that is a little clearer, and the diagrams and explanations here are also helpful now that I have had time to do some Google searches:

* https://ai-solutions.com/_help_Files/attitude_reference_frames.htm#achr_lvlhearthpointing
* https://spaceflight.nasa.gov/feedback/expert/answer/mcc/sts-105/08_14_08_01_18.html
* https://space.stackexchange.com/questions/33803/eci-to-lvlh-conversion

Using the matrix and rotation routines in `skyfield.functions`, it should be possible to put together a routine to compute such a reference frame across the vector arrays that Skyfield uses given a vector time. I’m not sure when I myself will have time to tackle it, but let’s go ahead and mark this issue an a feature request, as something that we would like to see added to Skyfield!

**Comment by matijabrumat** [comment]
Thanks. I will try to look into it and post some code if I make it working.

**Comment by matijabrumat** [comment]
@brandon-rhodes, I managed to produce the LVLH coordinate frame and calculate the view angles.  Minimal working example is published at stackexchange: https://space.stackexchange.com/questions/48796/local-observer-coordinate-system-at-satellite-panel-lvlh-coordinate-system

**Comment by brandon-rhodes** [comment]
Thanks for the working example, complete with code! I should have time to take a look within the next couple of weeks.
